src/library_webgl.js

```diff
@@ -2936,6 +3013,45 @@ var LibraryGL = {
     }
 #endif
 
+#if GL_EXPLICIT_UNIFORM_LOCATION
+#if GL_DEBUG
+    console.log('Input shader source: ' + source);
+#endif
+    // Remove comments and C-preprocess the input shader first, so that we can appropriately
+    // parse the layout location directives.
+    source = preprocess_c_code(remove_cpp_comments_in_shaders(source));
+#if GL_DEBUG
+    console.log('Shader source after preprocessing: ' + source);
+#endif
+    // Extract the layout(location = x) directives.
+    var regex = /layout\s*\(\s*location\s*=\s*([-\d]+)\s*\)\s*(uniform\s*\w*\s*(\w*))/g, explicitUniformLocations = {}, match;
```

Comment:
nit: it probably doesn't matter, but this seems not quite right, maybe something like this?:

```suggestion
    var regex = /layout\s*\(\s*location\s*=\s*([-\d]+)\s*\)\s*(uniform\s+\w+\s+(\w+))/g, explicitUniformLocations = {}, match;
```

Reply:
Thanks, updated.